openconnect-gp-okta

```diff
@@ -63,6 +87,60 @@ def okta_auth(s, domain, username, password, factor_priorities, totp_key):
                 code = input('SMS code: ')
                 r = post_json(s, url, {'stateToken': r['stateToken'], 'passCode': code})
                 break
+            if factor['factorType'] == 'webauthn' and not ignore_webauthn:
+                # Not using factor['_links']['verify']['href'] as multiple
+                # webauthn devices may exist and Okta has an alternative
+                # workflow which works with any webauthn device available by
+                # querying a generic URL.
+                url = 'https://{}/api/v1/authn/factors/webauthn/verify'.format(domain)
+                r = post_json(s, url, {'stateToken': r['stateToken']})
+                assert r['status'] == 'MFA_CHALLENGE'
+
+                device = next(list_devices(), None)
+                while device is None:
+                    click.echo("Please insert a suitable device if you wish to continue with webauthn MFA.")
+                    if click.confirm("Continue with webauthn MFA?"):
+                        device = next(list_devices(), None)
+                    else:
+                        break
+
+                if device is None:
+                    click.echo("Falling back to other MFA method.")
+                    url = r['_links']['prev']['href']
+                    post_json(s, url, {'stateToken': r['stateToken']})
```

Comment:
Should this assign to `r`? Or should we keep the original `r` around? Seems wrong to continue with the one from `webauthn/verify`...

Reply:
Thanks for pointing that out, it was an indeed a mistake. This should be fixed in the refactored version.